libs/coroutines/include/hpx/coroutines/coroutine.hpp

```diff
@@ -98,16 +97,35 @@ namespace hpx { namespace threads { namespace coroutines {
         }
 
 #ifdef HPX_HAVE_LIBCDS
-        typedef std::array<std::size_t, 3> cds_type;
-        cds_type get_libcds_data() const
+        std::size_t& get_libcds_data() const
         {
             return impl_.get_libcds_data();
         }
 
-        cds_type set_libcds_data(cds_type data)
+        std::size_t& set_libcds_data(std::size_t& data)
```

Comment:
What's your rationale of using references for the parameter and return value in this case?

Reply:
Oh, should be a typo, I meant const& everywhere, but I forgot to add it, will add soon.

Reply:
That makes it only slightly better. Why do you think you need references in the first place?

Reply:
Oh, because you commented here saying adding const references to avoid copy: https://github.com/STEllAR-GROUP/hpx/pull/4823#discussion_r453971287 and https://github.com/STEllAR-GROUP/hpx/pull/4823#discussion_r453972807

Reply:
@weilewei `sizeof(std::array<3, size_t>) == 3 * sizeof(std::array<3, size_t> const&)`, however `sizeof(size_t) == sizeof(size_t const&)`. Thus, if there is no size benefit of using references, then value semantics are preferable any day. Moreover, returning a reference from a function forces you to use a function-local static in the implementation which will create a data race as this variable would be shared in between threads.

Reply:
Thanks for the clarification, ok, then I will stick to value semantics then, pass by value and return by value.